File: src/test/java/com/bonepl/chromaleague/razer/sdk/effects/StaticEffectTest.java
package com.bonepl.chromaleague.razer.sdk.effects;

import com.bonepl.chromaleague.razer.RazerSDKClient;
import org.junit.jupiter.api.Test;

class StaticEffectTest {
    @Test
    void testStaticEffect() throws InterruptedException {
        try (RazerSDKClient razerSDKClient = new RazerSDKClient()) {
            razerSDKClient.createKeyboardEffect(new StaticKeyboardEffect(Color.WHITE));
            Thread.sleep(100);
            razerSDKClient.createKeyboardEffect(new StaticKeyboardEffect(Color.GREEN));
            Thread.sleep(100);
            razerSDKClient.createKeyboardEffect(new StaticKeyboardEffect(Color.RED));
            Thread.sleep(100);
            razerSDKClient.createKeyboardEffect(new StaticKeyboardEffect(Color.BLACK));
            Thread.sleep(100);
            razerSDKClient.createKeyboardEffect(new StaticKeyboardEffect(Color.BLUE));
            Thread.sleep(100);
            razerSDKClient.createKeyboardEffect(new StaticKeyboardEffect(Color.YELLOW));
            Thread.sleep(100);
            razerSDKClient.createKeyboardEffect(new StaticKeyboardEffect(Color.CYAN));
            Thread.sleep(100);
        }
    }
}

File: test/razer-sdk/effects/StaticEffect.test.ts
import {describe, test} from 'vitest';
import {ChromaNativeSDK} from '../../../src/razer-sdk/ChromaNativeSDK.js';
import {SimpleFrame} from '../../../src/razer-sdk/animation/SimpleFrame.js';
import {StaticColor} from '../../../src/razer-sdk/color/StaticColor.js';

/**
 * Integration test for static color effects on actual hardware.
 * Displays full-keyboard static colors cycling through the spectrum.
 */
describe('StaticEffect Integration', () => {
    test('testStaticEffect', async () => {
        const sdk = new ChromaNativeSDK();
        await sdk.connect();

        try {
            const colors = [
                StaticColor.WHITE, StaticColor.GREEN, StaticColor.RED,
                StaticColor.BLACK, StaticColor.BLUE, StaticColor.YELLOW, StaticColor.CYAN,
            ];

            for (const color of colors) {
                await sdk.createKeyboardEffect(new SimpleFrame(color));
                await new Promise(resolve => setTimeout(resolve, 500));
            }
        } finally {
            await sdk.close();
        }
    }, 30000);
});
